runtime/src/nonce_keyed_account.rs

```diff
@@ -277,9 +285,12 @@ pub fn authorize_nonce_account(
                 data.durable_nonce,
                 data.get_lamports_per_signature(),
             );
-            account.set_state(&Versions::new_current(State::Initialized(new_data)))
+            account.set_state(&Versions::new(
+                State::Initialized(new_data),
+                separate_domains,
+            ))
```

Comment:
`AuthorizeNonceAccount` does not change the value of nonce.
So this updated code is *preserving* domain and not bumping nonce out of blockhash domain because otherwise it needs to change the value of `durable_nonce` and that is a change of behavior.

Can we change the behavior here and change the nonce as well? cc @t-nelson

Reply:
I personally think we should preserve the version and the blockhash